sim/core/aura.go

```diff
@@ -910,11 +910,13 @@ func (caster *Unit) NewPartyAuraArray(makeAura func(*Unit) *Aura) AuraArray {
 }
 
 func (caster *Unit) NewEnemyAuraArray(makeAura func(*Unit, int32) *Aura) AuraArray {
-	auras := make([]*Aura, len(caster.Env.AllUnits))
-	for _, target := range caster.Env.AllUnits {
-		if target.Type == EnemyUnit {
-			auras[target.UnitIndex] = makeAura(target, caster.Level)
-		}
+	enemyUnits := FilterSlice(caster.Env.AllUnits, func(unit *Unit) bool {
+		return unit.Type == EnemyUnit
+	})
+
+	auras := make([]*Aura, len(enemyUnits))
+	for _, target := range enemyUnits {
+		auras[target.UnitIndex] = makeAura(target, caster.Level)
```

Comment:
This change limits the size of the aura array to only enemy units. This should be a bit more memory-performant as now the sim character and pets are not included

Reply:
Enemies are registered with their unit index so now the array size smaller then the last enemy index (due to player characters/pets). I actually wonder how is this working? Are enemies registered first so their unit indices start from 0 and it still works? Non the less i think the reason this was initialized with the full env units lenght is because we use unitIndex for indexing

Reply:
Ooooh good point. I totally overlooked that. I'll change it back